clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -157,6 +157,32 @@ static void optimizeCWD(CowCompilerInvocation &BuildInvocation, StringRef CWD) {
   }
 }
 
+/// Check a subset of invocation options to determine whether the current
+/// context can safely be considered as shareable.
+static bool areOptionsInSharedDir(CowCompilerInvocation &BuildInvocation,
+                                  const ArrayRef<StringRef> SharedDirs) {
+  const auto &HSOpts = BuildInvocation.getHeaderSearchOpts();
+  if (!isPathInSharedDir(SharedDirs, HSOpts.Sysroot))
```

Comment:
Should the first two checks be asserts since we derive sharable directories from sysroot and resource dir?  Or is there a case where they will differ?

Reply:
I can't think of this happening intentionally.